multi-node/aws/pkg/config/templates/cloud-config-controller

```diff
@@ -91,17 +93,18 @@ coreos:
       content: |
         [Unit]
         Description=decrypt kubelet tls assets using amazon kms
-        Before=kubelet.service
-        After=docker.service
-        Requires=docker.service
+        Before=kubelet.service flanneld.service
```

Comment:
Fair enough, I didn't consider that use case, though it's not like we have to outright ban anything.  Just preferring one unless there's a good reason to prefer the other would be fine.  UPS makes something like 90% right turns. 

Assuming I'm sold on this being better than mutating the kubelet dependency, I think we could get the same dependency injection effect with more clarity by using targets.  Kubelet should require something like network-provider.target and then having flanneld's RequiredBy set to the target.  This seems to be the pattern I see poking around /etc/systemd/system on my local arch install.

I just really dislike looking at a service and it not being obvious if it has any dependencies. 

Regardless of what whether or not we're following this injection pattern, it doesn't look like we're being consistent either way since flannel explicitly Requires decrypt, while kubelet doesn't.